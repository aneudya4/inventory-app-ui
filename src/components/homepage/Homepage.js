import React from 'react';
import { Link } from 'react-router-dom';
import img from './images/inv.jpeg';
// import Header from '../header/Header';
import './homepage.css';
const HomePage = () => {
  return (
    <>
      <div className='banner'>
        <div id='overlay'></div>
        <div className='banner-info'>
          <h2>
            Invy helps you track all your sales and your products in stock
          </h2>
          <p>
            Easily <span>add</span> and <span>delete</span> products, add orders
            and <span>keep track</span> of them all
          </p>
          <Link to='/accounts/register'>Register</Link>
          <Link to='/accounts/login'>Login</Link>
        </div>
        <div className='banner-img'>
          <img src={img} alt='banner' />
        </div>
      </div>
    </>
  );
};
export default HomePage;
