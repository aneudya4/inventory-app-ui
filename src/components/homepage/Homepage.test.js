import { shallow } from 'enzyme';
import Homepage from './Homepage';

describe('Test on the <Homepage/> Component', () => {
  it('renders Homepage without crashing', () => {
    const wrapper = shallow(<Homepage />);
    expect(wrapper).toMatchSnapshot();
  });

  it('renders Homepage with the Link element for Register', () => {
    const wrapper = shallow(<Homepage />);
    const RegisterLink = wrapper.find('Link').at(0);
    expect(RegisterLink.text()).toBe('Register');
    expect(RegisterLink.props('href')).toEqual({
      to: '/accounts/register',
      children: 'Register',
    });
  });

  it('renders Homepage with the Link element for Login', () => {
    const wrapper = shallow(<Homepage />);
    const LoginLink = wrapper.find('Link').at(1);
    expect(LoginLink.text()).toBe('Login');
    expect(LoginLink.props()).toEqual({
      to: '/accounts/login',
      children: 'Login',
    });
  });
});
