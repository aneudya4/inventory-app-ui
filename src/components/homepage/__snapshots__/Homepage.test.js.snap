// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test on the <Homepage/> Component renders Homepage without crashing 1`] = `
<Fragment>
  <div
    className="banner"
  >
    <div
      id="overlay"
    />
    <div
      className="banner-info"
    >
      <h2>
        Invy helps you track all your sales and your products in stock
      </h2>
      <p>
        Easily 
        <span>
          add
        </span>
         and 
        <span>
          delete
        </span>
         products, add orders and 
        <span>
          keep track
        </span>
         of them all
      </p>
      <Link
        to="/accounts/register"
      >
        Register
      </Link>
      <Link
        to="/accounts/login"
      >
        Login
      </Link>
    </div>
    <div
      className="banner-img"
    >
      <img
        alt="banner"
        src="inv.jpeg"
      />
    </div>
  </div>
</Fragment>
`;
